XmlFormatterOsIndependent/ViewModels/SettingsWindowViewModel.cs

```diff
@@ -0,0 +1,197 @@
+        private int themeMode;
+
+        /// <summary>
+        /// Check for updates on start
+        /// </summary>
+        public bool CheckUpdateOnStart
+        {
+            get => checkUpdateOnStart;
+            set => this.RaiseAndSetIfChanged(ref checkUpdateOnStart, value);
+        }
+        /// <summary>
+        /// Private check for updates on start
+        /// </summary>
+        private bool checkUpdateOnStart;
+
+        /// <summary>
+        /// The current updater to use
+        /// </summary>
+        public PluginMetaData Updater
+        {
+            get => updater;
+            set => this.RaiseAndSetIfChanged(ref updater, value);
+        }
+        /// <summary>
+        /// Private updater to use
+        /// </summary>
+        private PluginMetaData updater;
+        
+        /// <summary>
+        /// Index of the selected updater
+        /// </summary>
+        public int UpdaterIndex
+        {
+            get => updaterIndex;
+            set => this.RaiseAndSetIfChanged(ref updaterIndex, value);
+        }
+        /// <summary>
+        /// Private index of the selected updater
+        /// </summary>
+        private int updaterIndex;
+
+        /// <summary>
+        /// The setting scope of this application
+        /// </summary>
+        private readonly ISettingScope applicationScope;
+
+        /// <summary>
+        /// Create a new instance of this view
+        /// </summary>
+        /// <param name="view">The view for this model</param>
+        /// <param name="settingsManager">The settings manager to use</param>
+        /// <param name="pluginManager">The plugin manager to use</param>
+        public SettingsWindowViewModel(ViewContainer view, ISettingsManager settingsManager, IPluginManager pluginManager)
+            : base(view, settingsManager, pluginManager)
+        {
+            if (this.settingsManager == null || this.pluginManager == null)
+            {
+                return;
+            }
+            this.settingsManager.Load(settingsPath);
+            applicationScope = this.settingsManager.GetScope("Default");
+            List = pluginManager.ListPlugins<IUpdateStrategy>();
+
+            if (applicationScope == null)
+            {
+                applicationScope = new SettingScope("Default");
+                this.settingsManager.AddScope(applicationScope);
+            }
+            LoadSettings();
+        }
+
+        /// <summary>
+        /// Load all the settings
+        /// </summary>
+        private void LoadSettings()
+        {
+            AskBeforeClosing = GetSettingsValue<bool>("AskBeforeClosing");
+            CheckUpdateOnStart = GetSettingsValue<bool>("SearchUpdateOnStartup");
+            IDataCommand getTheme = new GetThemeCommand();
+            ExecuteCommand(getTheme, settingsManager);
+            ThemeMode = (int)getTheme.GetData<ThemeEnum>();
+
+            string updater = GetSettingsValue<string>("UpdateStrategy");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename 'updater' which hides the field with the same name.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5634024)